src/topp/DecoyDatabase.cpp

```diff
@@ -165,173 +187,215 @@ class TOPPDecoyDatabase :
-    String decoy_string_lower = decoy_string;
-    decoy_string_lower.toLower();
-    bool is_common = false;
-    for (const auto& a : DecoyHelper::affixes)
+
+ // Neighbor peptide search   
+    String neighbor_file = getStringOption_("Neighbor_Search:neighbor_in");
+ // Check if the neighbor file option is provided
+    if (! neighbor_file.empty())
     {
-      if ((decoy_string_lower.hasPrefix(a) && decoy_string_position_prefix) || (decoy_string_lower.hasSuffix(a) && !decoy_string_position_prefix))
+      String resolution = getStringOption_("Neighbor_Search:resolution");
+      double ion_tolerance = getDoubleOption_("Neighbor_Search:ion_tolerance");
+      double mass_tolerance = getDoubleOption_("Neighbor_Search:mass_tolerance");
+ // Create a ProteaseDigestion object for peptide digestion
+      ProteaseDigestion digestion;
+      String enzyme = getStringOption_("enzyme").trim();
+      if ((input_type == SeqType::protein) && ! enzyme.empty()) 
       {
-        is_common = true;
+          digestion.setEnzyme(enzyme); 
       }
-    }
-    // terminate, if decoy_string is not one of the allowed decoy strings (exit code 11)
-    if (!is_common)
-    {
-      if (getFlag_("force"))
+      
+ // Vector to hold all digested peptides from the input files
+      vector<AASequence> all_peptides;
+
+      for (Size i = 0; i < in.size(); ++i)
       {
-        OPENMS_LOG_WARN << "Force Flag is enabled, decoys with custom decoy string (not in DecoyHelper::affixes) will not be detected.\n";
+        //FASTAContainer<TFI_File> in_entries {in[i]};
+        f.readStart(in[i]);
+        while (f.readNext(entry))
+        {
+        
+          // Vector to hold the peptides generated from digestion
+          vector<AASequence> peptides;
+          // Digest the current entry sequence into peptides
+          digestion.digest(AASequence::fromString(entry.sequence), peptides);
+          // Add the generated peptides to the all_peptides vector
+          all_peptides.insert(all_peptides.end(), peptides.begin(), peptides.end());
+        }
       }
-      else
-      {
-        OPENMS_LOG_FATAL_ERROR << "Given decoy string is not allowed. Please use one of the strings in DecoyHelper::affixes as either prefix or suffix (case insensitive): \n";
-        return INCOMPATIBLE_INPUT_DATA;
+ // Load the neighbor peptides from the neighbor file
+      FASTAFile neighbor_fasta;
+      vector<FASTAFile::FASTAEntry> neighbor_entries;
+      neighbor_fasta.load(neighbor_file, neighbor_entries);
+      for (auto i = all_peptides.begin(); i != all_peptides.end(); ++i)
+      { 
+          {
+          // Find neighbor peptides for the current peptide
+            vector<FASTAFile::FASTAEntry> result = NeighborSeq::findNeighborPeptides(*i, neighbor_entries, mass_tolerance, ion_tolerance, resolution);
+            for (const auto& sub : result)
+            {
+            // Write the neighbor peptide to the output file
+                f.writeNext(sub);
```

Comment:
mh... now you have neighbor peptides in `out`, but there are no decoys yet.
This is fine, if the output of this DecoyDatabase step is again forwarded to a second DecoyDatabase which gets the relevant proteins and the neighbors from this step. But this feels a bit weird, since then the tool just creates neighbor peptides and has nothing to do with creating decoys (it should thus be a separate tool).